packages/ibm-products/src/components/Checklist/Checklist.js

```diff
@@ -0,0 +1,336 @@
+ * complete (e.g. tasks, walkthroughs, etc.) in order to learn about
+ * and/or onboard to a product.
+ */
+export let Checklist = React.forwardRef(
+  (
+    {
+      chartValue,
+      chartLabel,
+      checklistAriaLabel = defaults.checklistAriaLabel,
+      checklistToggleAriaLabel = defaults.checklistToggleAriaLabel,
+      className,
+      onClickViewAll = defaults.onClickViewAll,
+      onToggle = defaults.onToggle,
+      open = defaults.open,
+      showToggle = defaults.showToggle,
+      taskLists = defaults.taskLists,
+      theme = defaults.theme,
+      title,
+      viewAllLabel,
+      ...rest
+    },
+    ref
+  ) => {
+    const [isOpen, setIsOpen] = useState(open);
+    const listContainerId = useRef(uuidv4()).current;
+    const chartLabelId = useRef(uuidv4()).current;
+
+    // Don't use this test: {chartValue && chartLabel && (render...)}.
+    // If `chartValue` is 0 (zero) - a valid value - then it will render 0 and skip the remaining statement.
+    // Use this test instead: {chartLabelAndValue && (render...)}.
+    // The ChecklistChart component will validate `chartValue`.
+    const chartLabelAndValue = typeof chartValue === 'number' && chartLabel;
+
+    const handleClickToggle = () => {
+      setIsOpen((prevOpen) => !prevOpen);
+    };
+
+    const handleClickViewAll = () => {
+      onClickViewAll();
+    };
+
+    // If state changes, then trigger callback.
+    useEffect(() => {
+      onToggle(isOpen);
+    }, [isOpen, onToggle]);
+
+    // If the "open" prop is changed after initialization,
+    // then update internal state.
+    useEffect(() => {
+      setIsOpen(open);
+    }, [open]);
+
+    return (
+      <aside
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        aria-label={checklistAriaLabel}
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            [`${blockClass}__closed`]: !isOpen,
+          }
+        )}
+        ref={ref}
+        {...getDevtoolsProps(componentName)}
+      >
+        {(title || chartLabelAndValue) && (
+          <header className={`${blockClass}__header`}>
+            {chartLabelAndValue && (
+              <ChecklistChart
+                aria-labelledby={chartLabelId}
+                theme={theme}
+                value={chartValue}
+              />
+            )}
+
+            <div className={`${blockClass}__titles`}>
+              {title && <h2 className={`${blockClass}__title`}>{title}</h2>}
+
+              {chartLabelAndValue && (
+                <h3 id={chartLabelId} className={`${blockClass}__chart-label`}>
+                  {chartLabel}
+                </h3>
+              )}
+            </div>
+
+            {showToggle && (
+              <Button
```

Comment:
I think you could alternatively use the new `IconButton` in [Carbon v11](https://react.carbondesignsystem.com/?path=/docs/components-iconbutton--overview) here.

Reply:
Done.